app/src/main/java/com/guru/recognizetext/TRApplication.kt

```diff
@@ -0,0 +1,19 @@
+package com.guru.recognizetext
+
+import android.app.Application
+import com.guru.recognizetext.di.AppComponent
+import com.guru.recognizetext.di.AppModule
+import com.guru.recognizetext.di.DaggerAppComponent
+
+class TRApplication : Application() {
+    val appComponent: AppComponent by lazy {
+        DaggerAppComponent.builder()
+            .appModule(AppModule(this))
+            .build()
+    }
+
+    override fun onCreate() {
+        super.onCreate()
+        appComponent // Initialize Dagger
```

Comment:
Is this necessary? You can check by removing this line of code and removing onCreate as well and if the app still works fine, then we can remove the entire `onCreate` block.

Reply:
its works fine after removing entrie onCreate method